File: src/main/java/net/dontdrinkandroot/gitki/wicket/page/HistoryPage.java
package net.dontdrinkandroot.gitki.wicket.page;

import net.dontdrinkandroot.gitki.wicket.GitkiWebSession;
import net.dontdrinkandroot.gitki.wicket.component.bspanel.RevCommitPanel;
import net.dontdrinkandroot.gitki.wicket.dataprovider.HistoryDataProvider;
import net.dontdrinkandroot.wicket.bootstrap.component.pagination.AjaxPaginationPanel;
import org.apache.wicket.ajax.AjaxRequestTarget;
import org.apache.wicket.markup.html.WebMarkupContainer;
import org.apache.wicket.markup.repeater.Item;
import org.apache.wicket.markup.repeater.data.DataView;
import org.apache.wicket.model.IModel;
import org.apache.wicket.model.StringResourceModel;
import org.apache.wicket.request.mapper.parameter.PageParameters;
import org.eclipse.jgit.revwalk.RevCommit;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class HistoryPage extends DecoratorPage<Void>
{
    public HistoryPage(PageParameters parameters)
    {
        super(parameters);
        GitkiWebSession.get().assertAnonymousBrowsing(HistoryPage.class);
    }

    @Override
    protected IModel<String> createTitleModel()
    {
        return new StringResourceModel("gitki.history");
    }

    @Override
    protected void onInitialize()
    {
        super.onInitialize();

        WebMarkupContainer commitContainer = new WebMarkupContainer("commitContainer");
        commitContainer.setOutputMarkupId(true);
        this.add(commitContainer);

        DataView<RevCommit> commitView = new DataView<RevCommit>("commit", new HistoryDataProvider())
        {
            @Override
            protected void populateItem(Item<RevCommit> item)
            {
                item.add(new RevCommitPanel("detail", item.getModel()));
            }
        };
        commitView.setItemsPerPage(20);
        commitContainer.add(commitView);

        AjaxPaginationPanel pagination = new AjaxPaginationPanel("pagination", commitView)
        {
            @Override
            protected void onPageChanged(AjaxRequestTarget target)
            {
                target.add(this);
                target.add(commitContainer);
            }
        };
        this.add(pagination);
    }
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/wicket/page/HistoryPage.kt
package net.dontdrinkandroot.gitki.wicket.page

import net.dontdrinkandroot.gitki.wicket.component.commit.RevCommitPanel
import net.dontdrinkandroot.gitki.wicket.dataprovider.HistoryDataProvider
import net.dontdrinkandroot.gitki.wicket.getGitkiSession
import net.dontdrinkandroot.wicket.behavior.appendCssClass
import net.dontdrinkandroot.wicket.behavior.outputMarkupId
import net.dontdrinkandroot.wicket.bootstrap.component.pagination.AjaxPaginationPanel
import net.dontdrinkandroot.wicket.bootstrap.css.BootstrapCssClass
import net.dontdrinkandroot.wicket.kmodel.wrap
import net.dontdrinkandroot.wicket.markup.repeater.data.repeatingDataView
import org.apache.wicket.model.StringResourceModel
import org.apache.wicket.request.mapper.parameter.PageParameters

class HistoryPage(parameters: PageParameters) : DecoratorPage<Void>(parameters) {

    init {
        getGitkiSession().assertAnonymousBrowsing(HistoryPage::class.java)
    }

    override fun createTitleModel() = StringResourceModel("gitki.history").wrap()

    override fun onInitialize() {
        super.onInitialize()

        val commitView = repeatingDataView(
            "commit",
            HistoryDataProvider(),
            25,
            outputMarkupId(),
            appendCssClass(BootstrapCssClass.LIST_GROUP)
        ) { id, model ->
            RevCommitPanel(id, model, appendCssClass(BootstrapCssClass.LIST_GROUP_ITEM))
        }
        add(commitView)
        add(AjaxPaginationPanel("pagination", commitView))
    }
}